### PR TITLE
feat: [feat-secure-workspace-egress] GALI-933 and GALI-934 add UI changes for egress store request submission

### PR DESCRIPTION
**Pull request by MBtea**
Issue #, if available: GALI-933 and GALI-934

Description of changes:

- Add UI component to show egress store objects
- Add data egress store
- Add data egress controller

- [x] Have you successfully deployed to an AWS account with your changes?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?
- [ ] If new dependencies have been added, have they been pinned to specific versions? A: No dependencies are added.
- [ ] Is this change also required on the AWS Solution version? 
- [ ] Have you updated openapi.yaml if you made updates to API definition (including add, delete or update parameter and request data schema)? A: Only place holder are added in this PR and the APIs are not implemented yet. The openapi.yaml will be update when the APIs are implemented.
- [ ] If you had to run manual tests, have you considered automating those tests by adding them to [end-to-end tests](../main/end-to-end-tests/README.md)? A: The tests will be added when data egress store and APIs are implemented. No tests added in this PR.

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jn1119 — COMMENTED** [review]
Can you add a UI video or image for the request so that it is easier to visualize the change.

**Comment by MBtea** [comment]
Attaching screenshots for UI changes
UIEgressStoreButton
UIEgressStoreDetail

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/helpers/api.js [hunk 1]
```diff
@@ -338,6 +338,10 @@ function updateRegisteredAccount(accountId, data) {
   });
 }
 
+function getEgressStore(environmentId) {
+  return httpApiGet(`api/data-egress/${environmentId}`);
+}
+
```
Comment (jn1119) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/helpers/api.js:344]: This API doesn't exist yet I am guessing and there will be a separate PR on this?

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentEgressStoreDetail.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  renderEgressStoreTable() {
+    const pageSize = 5;
+    const showPagination = tempData.length > pageSize;
+    const egressStoreRequestSubmitted = this.egressStoreRequestSubmitted;
+    return (
+      <Segment placeholder className="mt2 mb2">
+        <ReactTable
+          data={tempData}
+          defaultSorted={[{ id: 'objectName', desc: true }]}
+          showPagination={showPagination}
```
Comment (jn1119) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentEgressStoreDetail.js:108]: Will the API handle pagination as well? I didn't see how we are passing the pagination input.

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentButtons.js [hunk 3]
```diff
@@ -9,6 +9,12 @@ import { displayError } from '@aws-ee/base-ui/dist/helpers/notification';
 
 import ScEnvironmentConnections from './ScEnvironmentConnections';
 import ScEnvironmentUpdateCidrs from './ScEnvironmentUpdateCidrs';
+import { enableEgressStore } from '../../../helpers/settings';
+import ScEnvironmentEgressStoreDetail from './ScEnvironmentEgressStoreDetail';
+
+const EGRESSING_STATUS_CODE = 'egressing';
+const WORKSPACE_TERMINATION_ERROR_MESSAGE =
+  'Termination is NOT allowed. Your data transfer from egress store is still in process. Please contact your Data Manager to confirm the data is transferred before you try to terminate the workspace again.';
```
Comment (nguyen102) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentButtons.js]: process => progress

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentEgressStoreDetail.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  }
+
+  getEgressStoreDetailStore() {
+    return this.props.scEnvironmentEgressStoreDetailStore;
+  }
+
+  handleSubmitEgressRequest = () => {
+    runInAction(() => {
+      this.egressStoreRequestSubmitted = !this.egressStoreRequestSubmitted;
+      console.log('egress request submitted');
```
Comment (nguyen102) [inline addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentEgressStoreDetail.js]: delete this?

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/helpers/api.js (hunk 1 above)
Comment (MBtea) [reply addons/addon-base-raas-ui/packages/base-raas-ui/src/helpers/api.js:344]: Yeah, this PR only contains UI components and placeholder for APIs and controller code. There are codes/comments in this PR should be remove after backend work is in place.

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentEgressStoreDetail.js (hunk 2 above)
Comment (MBtea) [reply addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentEgressStoreDetail.js:108]: Good point, since the backend code is not ready yet, I am not making them as paged yet. In the future PR, I will make changes after the backend is ready for this.

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentButtons.js (hunk 3 above)
Comment (MBtea) [reply addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentButtons.js]: Fixed in commit: b864b3246804d70395e6b3016ae038021b574474

File: addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentEgressStoreDetail.js (hunk 4 above)
Comment (MBtea) [reply addons/addon-base-raas-ui/packages/base-raas-ui/src/parts/environments-sc/parts/ScEnvironmentEgressStoreDetail.js]: Fixed in commit: b864b3246804d70395e6b3016ae038021b574474

**Review by jn1119 — COMMENTED** [review]
Can you double check why unit tests are failing?

**Comment by MBtea** [comment]
The unit testing is failing because of the lint issue. And I am putting some places holders in this commit and it should go away after I finish the backend functionality and then I can go back and finish the full UI functionality and the unit test should pass.

**Comment by MBtea** [comment]
new fix for lint issue has been pushed in commit: d9820be1934519b6b6aff20ab41275a0cd0b18ea

**Review by jn1119 — APPROVED** [review]
